WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigActivity.kt

```diff
@@ -1,17 +1,147 @@
+/*
+ * Copyright (C) 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.example.android.wearable.alpha.editor
 
 import android.os.Bundle
 import android.util.Log
+import android.view.View
 import androidx.activity.ComponentActivity
-import com.example.android.wearable.alpha.R
+import androidx.activity.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.example.android.wearable.alpha.data.watchface.ColorStyleIdAndResourceIds
+import com.example.android.wearable.alpha.databinding.ActivityWatchFaceConfigBinding
+import com.example.android.wearable.alpha.editor.WatchFaceConfigViewModel.Companion.MINUTE_HAND_LENGTH_DEFAULT_FOR_SLIDER
+import com.example.android.wearable.alpha.editor.WatchFaceConfigViewModel.Companion.MINUTE_HAND_LENGTH_MAXIMUM_FOR_SLIDER
+import com.example.android.wearable.alpha.editor.WatchFaceConfigViewModel.Companion.MINUTE_HAND_LENGTH_MINIMUM_FOR_SLIDER
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
-// TODO (codingjeremy): Add activity support to edit watch face (follow up PR).
+/**
+ * Allows user to edit certain parts of the watch face (color style, ticks displayed, minute arm
+ * length) by using the [WatchFaceConfigViewModel]. (All widgets are disabled until data is loaded.)
+ */
 class WatchFaceConfigActivity : ComponentActivity() {
+    private val viewModel: WatchFaceConfigViewModel by viewModels {
+        WatchFaceConfigViewModel.WatchFaceConfigViewModelFactory(
+            this@WatchFaceConfigActivity,
```

Comment:
Since the editor session requires an `Activity` (at least right now), we have to manage it within the `Activity`'s scope. If we pass it to the `ViewModel`, which will outlive the activity, we'll get memory leaks in the best case and in the worst case not work or crash since the editor is managing result contracts.